bot/cogs/config_verifier.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from discord.ext.commands import Cog
+
+from bot import constants
+from bot.bot import Bot
+
+
+log = logging.getLogger(__name__)
+
+
+class ConfigVerifier(Cog):
+    """Verify config on startup."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.bot.loop.create_task(self.verify_channels())
+
+    async def verify_channels(self) -> None:
+        """
+        Verify channels.
+
+        If any channels in config aren't present in server, log them in a warning.
+        """
+        await self.bot.wait_until_ready()
```

Comment:
We've had issues with the guild cache not being available when the READY signal was dispatched (discord.py only waits for 2 seconds which, during Discord API lag, is occasionally too short). To prevent that, we've implemented a custom [`wait_until_guild_available`](https://github.com/python-discord/bot/blob/c81a4d401ea434e98b0a1ece51d3d10f1a3ad226/bot/bot.py#L90) that ensures that the cache is actually loaded before doing things that rely on the cache being available. 

Since this validator relies entirely on the cache, it's a good idea to use that `wait_for` variant instead:

```suggestion
        await self.bot.wait_until_guild_available()
```